lib/evaluation/subjectIsHolderEvaluationHandler.ts

```diff
@@ -0,0 +1,74 @@
+import { HolderSubject, InputDescriptor, PresentationDefinition } from '@sphereon/pe-models';
+import jp from 'jsonpath';
+
+import { Status } from '../ConstraintUtils';
+import { VerifiableCredential, VerifiablePresentation } from '../verifiablePresentation';
+
+import { AbstractEvaluationHandler } from './abstractEvaluationHandler';
+import { HandlerCheckResult } from './handlerCheckResult';
+
+export class SubjectIsHolderEvaluationHandler extends AbstractEvaluationHandler {
+  public getName(): string {
+    return 'IsHolderEvaluation';
+  }
+
+  public handle(d: PresentationDefinition, p: VerifiablePresentation): void {
+    this.iterateOverInputCandidates(d, p);
+  }
+
+  private iterateOverInputCandidates(pd: PresentationDefinition, vp: VerifiablePresentation): void {
+    for (const vc of vp.getVerifiableCredentials().entries()) {
+      this.iterateOverInputDescriptors(pd, vc);
+    }
+  }
+
+  private iterateOverInputDescriptors(pd: PresentationDefinition, vc: [number, VerifiableCredential]): void {
+    const inputDescriptors: InputDescriptor[] = pd.input_descriptors;
+    const result = jp.query(vc[1], `$..credentialSubject.id`)[0];
```

Comment:
do we want the `jp` to look for `credentialSubject` recursively?

Reply:
Good question, I don't know if it uses BFS or DFS. So it may cause problems with nested vcs. However it works well with single vcs. Anyway, I'll change it.